Library/Homebrew/cask/audit.rb

```diff
@@ -580,13 +580,20 @@ def audit_rosetta
             add_error "No binaries in App: #{artifact.source}", location: cask.url.location if files.empty?
             system_command("lipo", args: ["-archs", files.first], print_stderr: false)
           when Artifact::Binary
-            system_command("lipo", args: ["-archs", path], print_stderr: false)
+            binary_path = path.to_s.gsub(cask.appdir, tmpdir)
+            system_command("lipo", args: ["-archs", binary_path], print_stderr: true)
           else
             add_error "Unknown artifact type: #{artifact.class}", location: cask.url.location
           end
 
-          unless result.success?
-            add_error "Failed to determine artifact architecture!", location: cask.url.location
+          # binary stanza can contain shell scripts, so we just continue if lipo fails.
+          next unless result.success?
+
+          odebug result.merged_output
+
+          unless /arm64|x86_64/.match?(result.merged_output)
```

Comment:
Will this blow up on universal binaries that include other architectures? If so: this should block merge.

Reply:
As long as it contains at least `arm64` or `x86_64` it'll pass.